guvnor-structure/guvnor-structure-api/src/main/java/org/guvnor/structure/repositories/Branch.java

```diff
@@ -0,0 +1,57 @@
+package org.guvnor.structure.repositories;
+
+import org.jboss.errai.common.client.api.annotations.MapsTo;
+import org.jboss.errai.common.client.api.annotations.Portable;
+import org.uberfire.backend.vfs.Path;
+
+import static org.uberfire.commons.validation.PortablePreconditions.checkNotNull;
```

Comment:
Branch, Repository, Project and Module objects are the "core" of all my changes.